src/App.js

```diff
@@ -1,85 +1,261 @@
-              <label htmlFor="todo-4">1234567890</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+class App extends React.Component {
+  state = {
+    todos: [],
+    newItemText: '',
+    checkedAll: false,
+    sortBy: '',
+    filterSelected: 'all',
+  }
+
+  componentDidMount() {
+    if (localStorage.getItem('todos') !== null) {
+      const todoItems = JSON.parse(localStorage.getItem('todos'));
+
+      this.setState(todoItems);
+    }
+  }
+
+  componentDidUpdate() {
+    const { todos } = this.state;
+
+    localStorage.setItem('todos', JSON.stringify({ todos }));
+  }
+
+  checkAll = ({ target }) => {
+    this.setState(({ todos, checkedAll }) => {
+      const newTodos = todos.map(todo => ({
+        ...todo,
+        done: target.checked,
+      }));
+
+      return {
+        todos: newTodos,
+        checkedAll: !checkedAll,
+      };
+    });
+  }
+
+  handleChange = ({ target }) => {
+    this.setState({
+      newItemText: target.value,
+    });
+  }
+
+  handleDelete = (id) => {
+    this.setState(({ todos }) => ({
+      todos: todos.filter(currentTodo => currentTodo.id !== id),
+    }));
+  };
+
+  toggleItem = (id) => {
+    this.setState(({ todos }) => {
+      const newTodos = todos.map((todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            done: !todo.done,
+          };
+        }
+
+        return todo;
+      });
+
+      const checkedAll = newTodos.every(todo => todo.done === true);
+
+      return {
+        todos: newTodos,
+        checkedAll,
+      };
+    });
+  }
+
+  handleAddItem = (event) => {
+    event.preventDefault();
+    this.addItem(this.state.newItemText);
+  }
+
+  changeText = (id, text) => {
+    this.setState(({ todos }) => {
+      const newItems = todos.map((todo) => {
+        if (todo.id === id && text !== '') {
+          return {
+            ...todo,
+            text,
+          };
+        }
+
+        if (todo.id === id && text === '') {
+          this.setState(() => ({
+            todos: todos.filter(currentTodo => currentTodo.id !== id),
+          }));
+        }
 
-      <footer className="footer">
-        <span className="todo-count">
-          3 items left
-        </span>
-
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
-
-          <li>
-            <a href="#/active">Active</a>
-          </li>
-
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
-
-        <button type="button" className="clear-completed">
-          Clear completed
-        </button>
-      </footer>
-    </section>
-  );
+        return todo;
+      });
+
+      return {
+        todos: newItems,
+      };
+    });
+  };
+
+  todosFilter = (todos, sortBy) => {
+    switch (sortBy) {
```

Comment:
```suggestion
    switch (filterBy) {
```
ты ж фильтруешь, не сортируешь